### PR TITLE
add Ddoc comments for TemplateParameter semantic routines

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

File: src/ddmd/semantic.d [hunk 1]
```diff
@@ -78,7 +78,17 @@ extern(C++) Type semantic(Type t, Loc loc, Scope* sc)
     return  v.result;
 }
 
-// Performs semantic on TemplateParamter AST nodes
+
+/************************************************
+ * Performs semantic on TemplateParameter AST nodes.
+ *
+ * Params:
+ *      tp = element of parameters[] to be semantically analyzed
```
Comment (JinShil) [inline src/ddmd/semantic.d]: parameters[] --> \`parameters\`

File: src/ddmd/semantic.d [hunk 2]
```diff
@@ -78,7 +78,17 @@ extern(C++) Type semantic(Type t, Loc loc, Scope* sc)
...
+ *      sc = context
+ *      parameters = array of TemplateParameters supplied to the TemplateDeclaration
```
Comment (JinShil) [inline src/ddmd/semantic.d]: TemplateParameters --> \`TemplateParameters\`
TemplateDeclaration --> \`TemplateDeclaration\`

File: src/ddmd/semantic.d [hunk 3]
```diff
@@ -78,7 +78,17 @@ extern(C++) Type semantic(Type t, Loc loc, Scope* sc)
...
+ * Returns:
+ *      true if no errors
```
Comment (JinShil) [inline src/ddmd/semantic.d]: true --> \`true\`

File: src/ddmd/templateparamsem.d [hunk 4]
```diff
@@ -1,5 +1,19 @@
...
+ *
+ * Copyright:   Copyright (c) 1999-2017 by Digital Mars, All Rights Reserved
+ * Authors:     $(LINK2 http://www.digitalmars.com, Walter Bright)
+ * License:     $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ * Source:      $(LINK2 https://github.com/dlang/dmd/blob/master/src/ddmd/templateparamsem.d, _templateparamsem.d)
+ */
+
 module ddmd.templateparamsem;
 
+// Online documentation: https://dlang.org/phobos/ddmd_templateparamsem.html
```
Comment (JinShil) [inline src/ddmd/templateparamsem.d]: That URL results in a 404

File: src/ddmd/templateparamsem.d [hunk 5]
```diff
@@ -99,6 +113,17 @@ extern (C++) final class TemplateParameterSemanticVisitor : Visitor
...
+/***********************************************
+ * Support function for performing semantic analysis on `TemplateAliasParameter`.
+ *
+ * Params:
+ *      loc = location (for error messages)
+ *      sc = context
+ *      o = object to run semantic() on, the `TemplateAliasParameter`s `specAlias` or `defaultAlias`
+ *      parameters = array of `TemplateParameters` supplied to the `TemplateDeclaration`
+ * Returns:
+ *      object resulting from running semantic() on `o`
```
Comment (JinShil) [inline src/ddmd/templateparamsem.d]: semantic() --> \`semantic()\`

**Review by JinShil — CHANGES_REQUESTED** [review]

File: src/ddmd/templateparamsem.d (hunk 4 above)
Comment (andralex) [reply src/ddmd/templateparamsem.d]: The link is correct, the file is forthcoming.

**Comment by andralex** [comment]
Thanks @JinShil. In order to save @WalterBright some time I've made the changes myself by committing into his branch.

**Review by andralex — APPROVED** [review]

**Comment by WalterBright** [comment]
Thanks @andralex ! (I didn't know I could make commits to other peoples' PRs. I'll have to ask you how to do that! It will save time.)

**Comment by wilzbach** [comment]
>  (I didn't know I could make commits to other peoples' PRs. I'll have to ask you how to do that! It will save time.)

GitHub supports this for almost a year now. It simply gives write access to the branch of the PR author.
If you look for more automation, there are dozens of guides on the net and I have shared my setup via the DWiki:

https://wiki.dlang.org/Guidelines_for_maintainers#Write_access_to_PRs

**Comment by andralex** [comment]
Yah, just click the "Edit" (pen) button in the file view and edit code straight in the browser.

**Pull request merged**